tests/test_genetic_value.py

```diff
@@ -0,0 +1,876 @@
+import numba
+import numpy as np
+import pandas as pd
+import pytest
+import tskit
+import tstrait
+
+
+@functools.lru_cache(maxsize=None)
+def all_trees_ts(n):
+    """
+    Generate a tree sequence that corresponds to the lexicographic listing
+    of all trees with n leaves (i.e. from tskit.all_trees(n)).
+
+    Note: it would be nice to include a version of this in the combinatorics
+    module at some point but the implementation is quite inefficient. Also
+    it's not entirely clear that the way we're allocating node times is
+    guaranteed to work.
+    """
+    tables = tskit.TableCollection(0)
+    for _ in range(n):
+        tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+    for j in range(1, n):
+        tables.nodes.add_row(flags=0, time=j)
+
+    L = 0
+    for tree in tskit.all_trees(n):
+        for u in tree.preorder()[1:]:
+            tables.edges.add_row(L, L + 1, tree.parent(u), u)
+        L += 1
+    tables.sequence_length = L
+    tables.sort()
+    tables.simplify()
+    return tables.tree_sequence()
+
+
+class Test_output:
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
+    def test_output_single(self, num_ind):
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        ts = msprime.sim_ancestry(num_ind, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1)
+        trait_df = tstrait.sim_trait(
+            ts=ts,
+            num_causal=5,
+            model=model,
+            alpha=0,
+            random_seed=1,
+        )
+        genetic_df = tstrait.calculate_genetic_value(ts, trait_df)
+
+        assert len(genetic_df) == num_ind
+        assert genetic_df.shape[1] == 3
+        assert np.array_equal(genetic_df["individual_id"], np.arange(num_ind))
+        assert len(genetic_df["genetic_value"]) == num_ind
+        assert np.array_equal(genetic_df["trait_id"], np.zeros(num_ind))
+
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
+    def test_output_multiple(self, num_ind):
+        mean = np.array([0, 0, 0])
+        var = np.array([1, 9, 16])
+        cor = np.array([[1, 0.1, 0.2], [0.1, 1, 0.3], [0.2, 0.3, 1]])
+        model = tstrait.trait_model(
+            distribution="multi_normal", mean=mean, var=var, cor=cor
+        )
+        ts = msprime.sim_ancestry(num_ind, sequence_length=100_000, random_seed=1)
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=1)
+        trait_df = tstrait.sim_trait(
+            ts=ts,
+            num_causal=5,
+            model=model,
+            alpha=0,
+            random_seed=1,
+        )
+        genetic_df = tstrait.calculate_genetic_value(ts, trait_df)
+
+        assert len(genetic_df) == num_ind * 3
+        assert genetic_df.shape[1] == 3
+        assert len(genetic_df["individual_id"]) == num_ind * 3
+        assert len(genetic_df["genetic_value"]) == num_ind * 3
+        assert len(genetic_df["trait_id"]) == num_ind * 3
```

Comment:
I think it would make sense here to assign a variable for the number of traits: `num_traits = cor.shape[0]` and use this variable for the tests. Makes them much more readable.